### DISCUSSION TITLE
WalletConnect Relay

### DISCUSSION THREAD

**Discussion by C00K1E-dev** [WalletConnect/walletconnect-monorepo#3442] [Developer Support] [answered]
So 3 days ago my in development dapp worked fine now im having these errors in console :
 ws.js:90 WebSocket connection to 'wss://relay.walletconnect.com/?auth=eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJkaWQ6a2V5Ono2TWtlbjRYeG1Db011dml5RnFMM1JOSEpRemFYNEdCVTVkTGdCbTgyb1Y3U3YzWSIsInN1YiI6ImNmYTJhNTBmZGNlNTdlYTE3NjVkMmQzYzc1YzA1ZWM0MmEwMWU4MGY1ZmMyYjUwMDdhNjkwNTU5NWNmMjI5NzIiLCJhdWQiOiJ3c3M6Ly9yZWxheS53YWxsZXRjb25uZWN0LmNvbSIsImlhdCI6MTY5MTg1NTg3NiwiZXhwIjoxNjkxOTQyMjc2fQ.DN3CmU8Z1BqHoCMcU9Het9Scm3W5oheMzntZmkj7rMf8DrMTZgpoSRPuxP8v4VZNOTsXwkzk1RmT6WSTgDISAQ&projectId=aca932c97e3f9bc59a1636dc1aeae670&ua=wc-2%2Fjs-2.9.0%2Fandroidos-chrome-115.0.0%2Fbrowser%3Alocalhost%3A3000&useOnCloseEvent=true' failed: Invalid frame header
eval @ ws.js:90
register @ ws.js:88
open @ ws.js:36
open @ provider.js:91
connect @ provider.js:14
eval @ index.es.js:1
transportOpen @ index.es.js:1
init @ index.es.js:1
Refused to frame 'https://verify.walletconnect.com/' because an ancestor violates the following Content Security Policy directive: "frame-ancestors https://*.mintandwin.com https://mintandwin.com".      

  I repeat 3 days ago it worked fine with no errors in console .... what update did you guys made?? GOLDEN RULE IN PROGRAMMING : IF IT WORKS DONT TOUCH IT !!

**Comment by C00K1E-dev** [comment]
WebSocket Connection Errors:
It appears that your website is trying to establish WebSocket connections to the 'wss://relay.walletconnect.com/' server, but there is an issue with the frame header in the response received from the server. These errors are often on the server side and might not be something you can directly fix in your code. You might want to check with the server maintainers or support team of 'walletconnect.com' to see if they are aware of any issues on their end.

Content Security Policy (CSP) Issue:
The error "Refused to frame 'https://verify.walletconnect.com/' because an ancestor violates the following Content Security Policy directive" indicates that the external site 'https://verify.walletconnect.com/' is attempting to load content into a frame on your website, but your CSP settings do not allow it.

**Comment by C00K1E-dev** [comment]
I just checked ChainGPT website they also use web3modal for connection on their site . I opened the console and the same errors are showing .... pancakeswap also ...  i checked opensea NO errors in console ... are you guys at walletconnect support  aware of this errors ?? will the errors  be resolved ?? @glitch-txs

**Reply by glitch-txs** [reply]
Weird, works fine on my end. What browser/device are you using? Can you try somewhere else? Have you installed an extension that might interfere?
You can check WalletConnect servers status [here](https://status.walletconnect.com/)

**Comment by C00K1E-dev** [comment]
![chaingpt](https://github.com/WalletConnect/walletconnect-monorepo/assets/122067978/263495cf-7099-451a-9096-2795f0b55f50)
![MAWerrors](https://github.com/WalletConnect/walletconnect-monorepo/assets/122067978/3f0ae494-e201-49df-8e11-48c6dabd15de)
![opensea](https://github.com/WalletConnect/walletconnect-monorepo/assets/122067978/028fb7d3-c080-4ae9-9314-86db4467991f)
![pancakeswap](https://github.com/WalletConnect/walletconnect-monorepo/assets/122067978/940349bc-38fb-4956-98f5-099ff33d2350)


@glitch-txs  its strange is it just for me local ??? i mean other websites that dont use web3modal doesnt throw these errors ... Opensea and rarible for example ... but like you can see in the screenshots websites that use web3modal to connect throw these errors ...  like i said in the case of my website 3 days ago everything was fine no errors ... i havent installed new apps or extensions at all ... windows updates yes i made however i dont think thats from windows updates ... well i dont want to make a fuss but since an external website  want to load a frame in my website thats blocked by CSP its a bit concerning ... that frame can contain malicious things if it was blocked by CSP .... clickjacking and stuffs

**Reply by glitch-txs** [reply]
We'll investigate it, sending this to the team.

**Reply by C00K1E-dev** [reply]
@glitch-txs  Awesome ! Thanks for the replies also !  Please keep me updated regarding  the topic ...

**Comment by n1c01a5** [comment]
Same issue

```
Refused to frame 'https://verify.walletconnect.com/' because an ancestor violates the following Content Security Policy directive: "frame-ancestors https://*.feature.sh https://feature.sh".
```

**Comment by glitch-txs** [comment]
are you guys using VPN or any specific browser configuration?

**Reply by C00K1E-dev** [reply]
No I am not using vpn nor browser settings 
As I said everything was fine before the errors started showing like 1 week ago

**Reply by glitch-txs** [reply]
Could you try on another browser?

**Reply by C00K1E-dev** [reply]
Right now I'm at a customer installing some software so right now it's like a random user would access my website so other browser other pc other internet provider console is full errors ![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/122067978/c584bec1-2ad6-409c-9857-9a1918fd397b)

**Reply by glitch-txs** [reply]
can you try creating a new project id for your project?  or try emptying you allow list might be related

Does this happen in other platforms for your customer?

**Comment by coindegen** [comment, marked as answer]
I'm getting the same error in my console when I work locally (localhost:4000):

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/87461192/8962c5f1-3b4d-4793-b891-41892fd252a1)

However, the error disappears on the deployed version of my site. That's because your CSP directives are based on my verified domain: 
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/87461192/656aca9c-6c18-462b-a2e3-2fc8fe58bafa)

<img width="232" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/87461192/3105ca44-439a-486d-9551-bab338b7428c">

Perhaps there is a way to relax your content security policy for users testing locally?

**Reply by glitch-txs** [reply]
@coindegen can you try on localhost:3000?

**Reply by coindegen** [reply]
The port doesn't seem to make a difference, I see the error on both 3000 and 4000.

However, it doesn't seem to hamper any functionality locally, so I'm just ignoring it during dev.

**Comment by C00K1E-dev** [comment]
That's the right answer @coindegen    I have found that also last night while trying to resolve the thing... 
The deployed version doesn't throw  errors anymore just on localhost and that's because of verified domains and permissions on wallet connect cloud account ... topic can be closed I guess

**Comment by BLOCKCHAINDEVWEB** [comment]
Same issue on Google Chrome, origin (localhost:3000) is not allowed to display content from 'https://verify.walletconnect.com/' due to violation of 'frame-ancestors' directive '.

```txt
Refused to frame 'https://verify.walletconnect.com/' because an ancestor violates the following Content Security Policy directive: "frame-ancestors...
```

This exception was not always present, but the domain switch between development/production and configuration change must have meant that the server responsible for the 'frame-ancestors' directive could no longer interpret a local domain.

After making sure I cleaned the browser's storage and data, adding a custom CSP header to my Nextjs Dapp by adding 'https://verify.walletconnect.com/', this didn't work. not solved the problem.

**Comment by basememara** [comment]
I get this and tons of other WC errors in dev, but not when deployed. Who cares right? Except makes the developer experience irritating and painful, very low confidence in deploying and unpleasant having me chasing ghosts, after all I'm wasting my time writing this comment in a thread I shouldn't even be in. Makes me a very grumpy dev...

**Reply by drake-nathan** [reply]
Agreed. There needs to be a solution. I spent 30 minutes on these errors. Clean consoles are important.

**Reply by glitch-txs** [reply]
Thanks for the feedback, we'll work to improve this 👍

**Reply by n1c01a5** [reply]
Same.

After set up the project id, I have this error in dev/prod but only one time.

**Comment by xyrus02** [comment]
Let me see if I get this right:

* WCv1 is not working anymore (WSS connection errors)
* To use WCv2 you have to register with your data on WC cloud portal and obtain a "project id"
* Because of your defunct CSP rules, integrating developers are unable to test their changes locally before going to a public stage (irrelevant if staging or prod)
* Fixing your defunct CSP is considered an "improvement" to be "worked on"
* The advertised alternative to cloud relays (self-hosted relays) are not implemented yet, or at least the public repository with a docker image suggests so in its readme file

Why again should we integrate our dApps with WalletConnect?

**Reply by glitch-txs** [reply]
there shouldn't be any CSP issue, if you have one could you share more information on this?

**Reply by xyrus02** [reply]
I am referring to the same CSP issue the other developers in the thread are talking about.

`Refused to frame 'https://verify.walletconnect.com/' because an ancestor violates the following Content Security Policy directive: "frame-ancestors...`

This message pops up in the console whenever I try to call `WalletConnectProvider.init` with any options. It is not possible to consume the Ethereum provider afterwards.

I still see this as a blocking issue based on my observations.

**Reply by glitch-txs** [reply]
@xyrus02 could you share a demo/reproducible example for us to check on this?

**Comment by luccamordente** [comment]
I'm seeing this error in test environments which doesn't share the same origin as my production environment. I'd expect that by properly setting _Allowed Domains_ it would go away, but it doesn't. As far as I can tell I can only verify one domain, right? Do you plan to cater for this use case? Thanks.

**Comment by dbhurley** [comment]
Still no resolution here?

**Reply by glitch-txs** [reply]
Should be fixed in the latest version, if you're using Wagmi this is 1.4.2 or greater.

**Reply by Ashoat** [reply]
I'm using Wagmi 1.4.3 and am still seeing this issue.

**Comment by yellowCrimsonGator** [comment]
Hello guys.
Facing same issue on localhost. The connection modal only gets open after several attempts.

**Reply by canpoyrazoglu** [reply]
Have you solved this? I haven't deployed yet but the modal only gets triggered in the second attempt. The first one results in a JS error.

**Reply by C00K1E-dev** [reply]
I have already replied here like 2 months ago : That's the right answer @coindegen I have found that also last night while trying to resolve the thing...
The deployed version doesn't throw errors anymore just on localhost and that's because of verified domains and permissions on wallet connect cloud account ... topic can be closed I guess
So guys you should log on wallet connect cloud and add the permissions for the localhost

**Reply by canpoyrazoglu** [reply]
@C00K1E-dev oh sorry I've missed that (I need more coffee :)) there's a confusion as it explicitly says that localhost is always whitelisted. but I'll give it another look.

**Reply by C00K1E-dev** [reply]
@canpoyrazoglu no worries... 
To be honest I didn't do any modifications in the wallet connect cloud account I just realised that the error is because of the whitelisting in the account so this is why I said the thread can be closed in my previous responses .... at the moment I still get that error on localhost but I know why and I just ignore it since on the deployed version everything is ok

**Reply by pde-rent** [reply]
Same thing here - mislead by the docs stating localhost is always whitelisted, leads to trust issues ahead of deployment

**Comment by nicolas-zozol** [comment]
frame-ancestor security is not a random thing. I'm not a specialist, but I think it avoids a site to embed your Dapp and prevents fishing bad experience.
We have cut it on our netlify. WC is not working today anymore but it was working before. I'm not sure if it's related to this error or not
